app/controllers/api/usage_limits_controller.rb

```diff
@@ -55,20 +57,14 @@ def destroy
   private
 
   def find_plan
-    @plan = find_application_plan || find_end_user_plan
+    @plan = find_application_plan
   end
 
   def find_application_plan
     id = params[:application_plan_id]
     current_account.application_plans.find(id) if id
   end
```

Comment:
I think we should do the following and remove the other method:
```ruby
def find_plan
  @plan = current_account.application_plans.find(params[:application_plan_id])
end
```
So we make this param mandatory and if it isn't there, it will raise a record not found and return a 404, which is better than the ugly error of "method not found for nil class" of not finding the plan because the id is not in the param and then doing `@plan.whatever_method` ass I see it is called for all methods because the plan is mandatory anyway.